### ISSUE TITLE
OneToOne relationship doens't work, and other questions

### ISSUE THREAD

**Issue by navy1978** [ghidoz/angular2-jsonapi#260]
Hi,
Not clear how to work with relationships and jsonapi2angular library.

Forst question is: Are BelongsTo and HasMany mandatory for all relationship ManyToOne?
Second question: What about OneToOne relationship? is belongsTo mandaroty ?

And can you please clarify what is wrong in the following example?

I made some test with ManyToOne cases adding the BelongsTo and HasMany and it works correctly when I ask server for inclusion so for example soemthing liek this:

 
`@JsonApiModelConfig({
    type: 'as'
})
export class A extends JsonApiModel {

    @HasMany()
    bs: B[];
}
@JsonApiModelConfig({
    type: 'bs'
})
export class B extends JsonApiModel {

	@Attribute()
	field

    @BelongsTo()
    a: A;
	

}
`
So to create this request:

GET /a?include=b

I used as suggested in the documentation:

`this.datastore.findAll(A, {
    page: { size: 10, number: 1},
    include: 'b'
}).subscribe(
    (document) => {
        console.log(document.getMeta()); // metadata
        console.log(document.getModels()); // models
    }
);`

specifing also the way to mab the object:

`const config: DatastoreConfig = {
  baseUrl: 'http://localhost:8000/v1/',
  models: {
    data: A,
    b: B,
  }
}`
 
But I cannot make it work ir the relationship between A and B is oneToOne,

fo in this case I used exactly the same code but the following:

`export class A extends JsonApiModel {

    @Attribute()
    bs: B;
}`

But in that case the include doesn't work and B is not extracted by the response.
What I miss here?

Thank you

**Comment by hpawe01** [comment]
Yes, if you want to indicate, that an attribute in your model is a relationship, you should either use the `@BelongsTo` or `@HasMany` decorator (depending on the relationship).
For OneToOne relations you can check [this comment](https://github.com/ghidoz/angular2-jsonapi/issues/204#issuecomment-447008095)
